lib/fsm/follow.go

```diff
@@ -59,66 +61,82 @@ func (e *PlanRolledBackEvent) isTerminalEvent() bool { return true }
 
 // FollowOperationPlan returns a channel that will be receiving phase updates
 // for the specified plan.
+// The returned channel must be served to avoid blocking the producer as
```

Comment:
Moved the whole logic of the `FollowOperationPlan` inside of the goroutine to do away with buffered channel and avoid the issue in https://github.com/gravitational/gravity/pull/2426 entirely. All uses of the API were already servicing the channel so there was no need for the buffer.